### ISSUE TITLE
Would it be possible to use filter instead of whitelist?

### ISSUE THREAD

**Issue by crynobone** [darsain/laravel-console#1]
I understand the reason to have whitelist location, however Scribe bundle for example use filter as an easy authentication middleware to authenticate everything that route to console.

See https://github.com/daylerees/laravel-scribe

**Comment by darsain** [comment]
If I understand correctly, you want there to be a `console::console.filter` config with a value of `whitelist` that would be used in:

``` php
Route::group(array('before' => Config::get('console.filter', Config::get('console::console.filter'))), function () {
    // routes
})
```

so you could replace it with your own, right?

**Comment by crynobone** [comment]
It is possible, however configuration would be lost after upgrade :)

## 

Mior Muhammad Zaki
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by darsain** [comment]
And that's why all config is loaded from `console.option`, and falls back to `console::console.option` (as it is illustrated in my code snippet above). If you are changing the config in `bundles/console/config/console.php`
, read the [config documentation section](https://github.com/Darsain/laravel-console#config) that explains how to change the config and keep it there regardless of upgrades.

And is this what you are asking for anyway? I'm not sure if I understand you, because _Would it be possible to use filter instead of whitelist?_ doesn't make sense, as whitelist IS a filter.

So let me rephrase: Are you asking to have an option to use a CUSTOM filter for console routes? I.e. the very thing illustrated in my code snipped above?

**Comment by darsain** [comment]
I hope this solves your problem. But be aware that having this bundle on production servers is dangerous.
